README.md

```diff
@@ -115,13 +115,63 @@ Note that `called` can be used as a chainable method.
 
 #### .with
 
-Assert that a spy has been called with a given argument.
+Assert that a spy has been called with a given argument at least once,
+even if more arguments were provided.
 
 ```js
 expect(spy).to.have.been.called.with('foo');
 spy.should.have.been.called.with('foo');
 ```
 
+Will pass for `spy('foo')`, `spy('foo', 'bar')`, `spy(); spy('foo')`,
+etc.
+
+Can be used for multiple arguments:
+
+```js
+spy('foo');
+spy('bar');
+expect(spy).to.have.been.called.with('bar', 'foo');
+spy.should.have.been.called.with('bar', 'foo');
+```
+
+#### .with.exactly
+
+Similar to .with, but will pass only if the list of arguments is
+exactly the same as the one provided.
+
+```js
+expect(spy).to.have.been.called.with.exactly('foo', 'bar');
+spy.should.have.been.called.with.exactly('foo', 'bar');
+```
```

Comment:
Here you could provide the `spy()` calls within the code, like you've done above, to illustrate when these assertions pass